### PR TITLE
Add "proto_deps" to Metadata generation

### PR DESCRIPTION
**Pull request by geigerj**
This field already specifies the package's proto package
dependencies in the Artman config. Reading it in allow metadata
generation to specify these additional dependencies, e.g., PubSub's
dependency on the IAM proto package.

Note that because this field was previously only used by
Java/Packman, some additional configuration to specify the
language-specific package name is added via the "name_override"
field.

To be merged simultaneously with: https://github.com/googleapis/artman/pull/138, https://github.com/googleapis/googleapis/pull/248

### CODE REVIEW

**Comment by geigerj** [comment]
cc: @bjwatson -- this PR fixes the last Python hand-edit.

File: src/test/java/com/google/api/codegen/testdata/library_pkg.yaml [hunk 1]
```diff
@@ -6,6 +6,9 @@ author: Google, Inc.
 email: googleapis-packages@google.com
 homepage: https://github.com/googleapis/googleapis
 license: Apache-2.0
+proto_gen_pkg_deps:
```
Comment (shinfan) [inline src/test/java/com/google/api/codegen/testdata/library_pkg.yaml]: `proto_gen_pkg_deps` is a bit wordy. How about just `proto_gen_deps` or `proto_deps`

File: src/main/java/com/google/api/codegen/config/PackageMetadataConfig.java [hunk 2]
```diff
@@ -204,6 +201,42 @@ public static PackageMetadataConfig createFromString(String yamlContents) {
...
+      Map<String, Object> configMap) {
+    Map<TargetLanguage, Map<String, VersionBound>> packageDependencies = new HashMap<>();
+    List<String> packages = (List<String>) configMap.get("proto_gen_pkg_deps");
+
+    for (String packageName : packages) {
+      Map<String, Map<String, String>> config =
+          (Map<String, Map<String, String>>) configMap.get(packageName + "_version");
+      if (config == null) {
+        throw new IllegalArgumentException(
+            "'" + packageName + "' in proto_gen_pkg_deps was " + "not found in dependency list.");
```
Comment (shinfan) [inline src/main/java/com/google/api/codegen/config/PackageMetadataConfig.java]: "'" + packageName + "' in proto_gen_pkg_deps was not found in dependency list."

File: src/main/java/com/google/api/codegen/transformer/PackageMetadataTransformer.java [hunk 3]
```diff
@@ -27,6 +34,16 @@
...
+    return generateMetadataView(packageConfig, model, template, outputPath, language, null);
+  }
+
+  public PackageMetadataView.Builder generateMetadataView(
+      PackageMetadataConfig packageConfig,
+      Model model,
+      String template,
+      String outputPath,
+      TargetLanguage language,
+      Set<String> whitelistedDependencies) {
```
Comment (shinfan) [inline src/main/java/com/google/api/codegen/transformer/PackageMetadataTransformer.java]: It is not very clear that what this `whitelistedDependencies` does. Add some comments.

Comment (geigerj) [inline src/main/java/com/google/api/codegen/transformer/PackageMetadataTransformer.java]: Done

File: src/main/java/com/google/api/codegen/config/PackageMetadataConfig.java (hunk 2 above)
Comment (geigerj) [inline src/main/java/com/google/api/codegen/config/PackageMetadataConfig.java]: Done

File: src/test/java/com/google/api/codegen/testdata/library_pkg.yaml (hunk 1 above)
Comment (geigerj) [inline src/test/java/com/google/api/codegen/testdata/library_pkg.yaml]: Changed to `proto_deps` (and in googleapis/artman).

(I originally used `proto_gen_pkg_deps` because that was already the name of the field in the artman config.)
